File: appServer/DestinyLimoServer/Models/UserExamAnswer.cs
namespace DestinyLimoServer.Models
{
    //INSERT INTO `destiny-limo-db`.`user_exam_answers`
    /*exam_question_id`,
    `exam_id`,
    `mcq_id`,
    `choice_3_answer`,
    `choice_4_answer`,
    `choice_2_answer`,
    `choice_1_answer`)*/

    // use exact column names

    public class UserExamAnswer
    {
        public int exam_id { get; set; }  // exam_id
        public int exam_question_id { get; set; }  // exam_question_id
        public int mcq_id { get; set; }  // mcq_id
        public bool choice_1_answer { get; set; }  // choice_1_answer
        public bool choice_2_answer { get; set; }  // choice_2_answer
        public bool choice_3_answer { get; set; }  // choice_3_answer
        public bool choice_4_answer { get; set; }  // choice_4_answer

        public bool attempted { get; set; }
        public bool is_correct { get; set; }
    }
}

File: ui/src/store/models/UserExamAnswer.ts
import { MaterialMCQ } from "./MaterialMCQ";

export interface UserExamAnswer {
    exam_id: number;          // exam_id
    exam_question_id?: number; // exam_question_id
    mcq_id: number;           // mcq_id
    choice_1_answer: boolean; // choice_1_answer
    choice_2_answer: boolean; // choice_2_answer
    choice_3_answer: boolean; // choice_3_answer
    choice_4_answer: boolean; // choice_4_answer
    attempted: boolean;       // attempted
    is_correct: boolean;      // is_correct
}
